File: pdf-viewer/src/main/java/com/github/pdfviewer/HashUtils.java
package com.github.pdfviewer;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class HashUtils {
    public static String getMD5Hash(String filePath) {
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            md.update(filePath.getBytes());
            byte[] digest = md.digest();
            StringBuilder sb = new StringBuilder();
            for (byte b : digest) {
                sb.append(String.format("%02x", b));
            }
            return sb.toString();
        } catch (NoSuchAlgorithmException e) {
            //DLog.handleException(e);
            return null;
        }
    }
}

File: pdf-viewer/src/main/java/com/github/pdfviewer/HashUtils.kt
package com.github.pdfviewer

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object HashUtils {
    @JvmStatic
    fun getMD5Hash(filePath: String): String? {
        try {
            val md = MessageDigest.getInstance("MD5")
            md.update(filePath.toByteArray())
            val digest = md.digest()
            val sb = StringBuilder()
            for (b in digest) {
                sb.append(String.format("%02x", b))
            }
            return sb.toString()
        } catch (e: NoSuchAlgorithmException) {
            //DLog.handleException(e);
            return null
        }
    }
}
